src/monitor/group_state_machine.c

```diff
@@ -345,27 +345,31 @@ WalDifferenceWithin(AutoFailoverNode *secondaryNode,
 					AutoFailoverNode *otherNode, int64 delta)
 {
 	int64 walDifference = 0;
+	XLogRecPtr secondaryLsn = 0;
+	XLogRecPtr otherNodeLsn = 0;
 
-	if (otherNode == NULL)
+
+	if (secondaryNode == NULL || otherNode == NULL)
 	{
 		return true;
 	}
 
-	/* we want the delta which was reported most recently */
-	if (TimestampDifferenceExceeds(secondaryNode->walReportTime,
-								   otherNode->walReportTime, 0))
+	secondaryLsn = secondaryNode->currentLSN;
+	otherNodeLsn = otherNode->currentLSN;
+
+	if (secondaryLsn == 0 || otherNodeLsn == 0)
 	{
-		walDifference = otherNode->walDelta;
+		/* we don't have any data yet */
+		return false;
 	}
-	else
+
+	if (secondaryLsn > otherNodeLsn)
```

Comment:
I think this would be easier to read as

``` C
    walDifference = abs(otherNodeLsn - secondaryLsn);
```

But then I wonder why we accept that the LSN might be in any order in between the secondary and the primary. Certainly the primary LSN is supposed to always be greater than or equal to the secondary's LSN? If we keep using the absolute difference, then I think we need to explain why in a comment...